hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/HoodieTimelineArchiveLog.java

```diff
@@ -134,12 +161,199 @@ public boolean archiveIfRequired(HoodieEngineContext context) throws IOException
         LOG.info("No Instants to archive");
       }
 
+      if (config.getArchiveAutoMergeEnable()) {
+        mergeArchiveFilesIfNecessary(context);
+      }
       return success;
     } finally {
       close();
     }
   }
 
+  private void mergeArchiveFilesIfNecessary(HoodieEngineContext context) throws IOException {
```

Comment:
can we add a java doc comment on what all we do in this method.

Reply:
Added. Thanks a lot for your review